File: Edward Poot/src/QL/ast/expr/gt.go
package expr

import "ql/ast/visit"

type GT struct {
	Lhs, Rhs Expr
}

func (g GT) GetLhs() Expr {
	return g.Lhs
}

func (g GT) GetRhs() Expr {
	return g.Rhs
}

func (g GT) Eval() interface{} {
	return g.GetLhs().Eval().(int) > g.GetRhs().Eval().(int)
}

func (g GT) Accept(v visit.Visitor) interface{} {
	return v.Visit(g)
}

File: Kevin van den Bekerom/DSLQL/src/main/java/org/uva/sea/ql/ast/expr/logic/GT.java
package org.uva.sea.ql.ast.expr.logic;

import org.uva.sea.ql.ast.expr.BinaryExpr;
import org.uva.sea.ql.ast.expr.Expr;
import org.uva.sea.ql.ast.form.Context;
import org.uva.sea.ql.ast.visit.Visitor;
import org.uva.sea.ql.type.NumericalType;
import org.uva.sea.ql.type.Type;

public class GT extends BinaryExpr {
	
	public GT(Expr lhs, Expr rhs) {
		super(lhs, rhs);
	}
	
	@Override
	public Boolean eval() {
		return (Integer) lhs.eval() > (Integer) rhs.eval();
	}
	
	public void accept(Visitor visitor, Object context) {
		visitor.visit(this, context);
	}

	@Override
	public String toString() {
		return super.toString() + " (>)";
	}
	
	@Override
	public Type getType(Context context) {
		return new NumericalType();
	}
}
